python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -74,6 +73,7 @@ class Node(object):
     thread-safe usage.
     """
     __metaclass__ = abc.ABCMeta
+    __slots__ = ('_index_cache_dict', '_parent', '_read_lock', '_write_lock', '_dirty', '_registry', '_data_dict')
```

Comment:
We should be very careful about defining `__slots__`. It can be a very good performance improvement but when I checked in the past it made no real difference. It also prevents us from ever dynamically adding attributes to objects of this type.